ui/src/components/shared/CredentialRevokeModal.tsx

```diff
@@ -0,0 +1,72 @@
+import { Modal, Space, Typography, message } from "antd";
+import { useState } from "react";
+
+import { revokeCredential } from "src/adapters/api/credentials";
+import { ReactComponent as IconClose } from "src/assets/icons/x.svg";
+import { useEnvContext } from "src/contexts/env";
+import { Credential } from "src/domain";
+
+export function CredentialRevokeModal({
+  credential,
+  onClose,
+  onRevoke,
+}: {
+  credential: Credential;
+  onClose: () => void;
+  onRevoke: () => void;
+}) {
+  const env = useEnvContext();
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+
+  const { revNonce: nonce } = credential;
+
+  const handleRevokeCredential = () => {
+    setIsLoading(true);
+
+    void revokeCredential({ env, nonce }).then((response) => {
+      if (response.isSuccessful) {
+        onClose();
+        onRevoke();
+
+        void message.success({
+          content: (
+            <Space align="start" direction="vertical" style={{ width: "auto" }}>
+              <Typography.Text strong>
+                Revocation requires issuer state to be published
+              </Typography.Text>
+              <Typography.Text type="secondary">
+                Publish issuer state now or bulk publish with other actions.
+              </Typography.Text>
+            </Space>
+          ),
+        });
+      } else {
+        setIsLoading(false);
+
+        void message.error(response.error.message);
+      }
+    });
+  };
+
+  return (
+    <Modal
+      centered
+      closable
+      closeIcon={<IconClose />}
+      maskClosable
+      okButtonProps={{ danger: true, loading: isLoading }}
+      okText="Revoke credential"
+      onCancel={onClose}
+      onOk={handleRevokeCredential}
+      open
+      title="Are you sure you want to revoke this credential?"
+    >
+      <Space direction="vertical">
+        <Typography.Text type="secondary">
+          Revoking of a credential must be accompanied by publishing of issuer state in order for
+          the action to be recorded. This action cannot be undone.
```

Comment:
Probably worth adding a linebreak for "This action cannot be undone.", as in the [designs](https://www.figma.com/file/BE3tUbvtSB2EqmJsybpeFC/Polygon-ID-Platform?node-id=4680-185509&t=RKB7fxpGkonHTQNY-0).

Reply:
![image](https://user-images.githubusercontent.com/19190248/231745628-621c8057-6420-4875-b506-a88945403eaa.png)

Reply:
Sorry, I commented this in the wrong place. It's for this modal.

![Screenshot from 2023-04-13 14-55-02](https://user-images.githubusercontent.com/2691584/231765158-00807df1-c514-4957-a9b0-3bd1251ab8b6.png)

Reply:
Nevermind.